.pre-commit/.just

```diff
@@ -19,6 +19,16 @@ cargo-nightly-fmt: _check-cargo
         exit 1
     }
 
+
+check-docker: _check-docker
+    #!/usr/bin/env bash
+
+    if ! docker info > /dev/null 2>&1; then
+        echo "❌ Docker is not running. Please start Docker and try again." >&2
+        exit 1
+    fi
```

Comment:
IMHO, we shouldn't enforce developers to use Docker when there are multiple Docker-compatible options available (ex: `podman`). With podman, for example, it's possible to configure the `DOCKER_HOST` in a way that it's transparent for `testcontainers` where it's running the containers.